library/std/src/net/endian.rs

```diff
@@ -0,0 +1,37 @@
+//! Define internal endianness primitives for networking.
+
+/// A Big Endian u32.
+///
+/// Mostly used as inner type of IpvAddr.
+#[repr(transparent)]
+#[derive(Copy, Clone, PartialEq)]
+pub(super) struct BeU32(u32);
+
+impl BeU32 {
+    /// The argument `n` is interpreted as big endian.
+    ///
+    /// Caller has to ensure this invariant are met.
+    pub const fn from_be(n: u32) -> Self {
+        Self(n)
+    }
+
+    /// Interprets `arr` in big endian order.
```

Comment:
Maybe add a comment saying that `arr` is interpreted in native endianess? Or maybe even make that part of the method name?